packages/node_modules/@ciscospark/media-engine-webrtc/test/unit/spec/engine.js

```diff
@@ -0,0 +1,776 @@
+
+            return expectEvent(20000, `negotiationneeded`, engine);
+          })
+          .then(() => engine.createOffer())
+          .then(mockAnswer)
+          .then((answer) => engine.acceptAnswer(answer))
+          .then(() => {
+            assert.calledWith(spy, {
+              video: {
+                facingMode: {
+                  ideal: `environment`
+                }
+              }
+            });
+
+            assertLocalMedia(engine, `sendrecv`, `sendrecv`);
+          });
+      });
+
+      it(`adds a new track to an inprogress connection`, () => {
+        assert.notCalled(spy);
+        engine.setMedia(`audio`, `sendrecv`);
+        return engine.createOffer()
+          .then((offer) => {
+            return mockAnswer(offer);
+          })
+          .then((answer) => engine.acceptAnswer(answer))
+          .then(() => {
+            assert.calledOnce(spy);
+
+            assertLocalMedia(engine, `sendrecv`, `inactive`);
+            engine.setMedia(`video`, `sendrecv`, {
+              facingMode: {
+                ideal: `environment`
+              }
+            });
+
+            return expectEvent(20000, `negotiationneeded`, engine);
+          })
+          .then(() => engine.createOffer())
+          .then(mockAnswer)
+          .then((answer) => engine.acceptAnswer(answer))
+          .then(() => {
+            assert.calledWith(spy, {
+              video: {
+                facingMode: {
+                  ideal: `environment`
+                }
+              }
+            });
+
+            assertLocalMedia(engine, `sendrecv`, `sendrecv`);
+          });
+      });
+    });
+  });
+});
+
+function assertLocalKind(kind, direction, engine) {
+  const tracks = engine.localMediaStream.getTracks().filter((t) => t.kind === kind);
+  const senders = engine.pc.getSenders().filter((s) => s.track.kind === kind);
+  if (direction.includes(`send`)) {
+    assert.lengthOf(tracks, 1, `there is 1 local ${kind} track`);
+    assert.lengthOf(senders, 1, `local ${kind} is ${direction}`);
+    assert.isTrue(tracks[0].enabled, `the local ${kind} track is enabled`);
+    assert.isTrue(senders[0].track.enabled, `local ${kind}is ${direction}`);
+  }
+  else {
+    try {
+      assert.lengthOf(tracks, 0, `there are 0 local ${kind} tracks`);
+      assert.lengthOf(senders, 0, `there are 0 ${kind} senders`);
+    }
+    catch (err) {
+      assert.lengthOf(tracks, 1, `there is 1 local ${kind} track`);
+      assert.lengthOf(senders, 1, `local ${kind} is ${direction}`);
+      assert.isFalse(tracks[0].enabled, `the local ${kind} track is not enabled`);
+      assert.isFalse(senders[0].track.enabled, `local ${kind}is ${direction}`);
+    }
+  }
+}
+
+function assertRemoteKind(kind, direction, engine) {
+  const tracks = engine.remoteMediaStream.getTracks().filter((t) => t.kind === kind);
+  // const pcTracks = engine.pc.getRemoteStreams[0] && engine.pc.getRemoteStreams[0].getTracks().filter((t) => t.kind === kind);
```

Comment:
`pcTracks` tests supposed to be commented?

Reply:
yea, i think maybe there's a comment in assertLocalKind that didn't make it here...